src/animal/animal.controller.ts

```diff
@@ -76,27 +87,37 @@ export class AnimalController {
   @Delete(':id')
   @ApiParam({ name: 'id', description: 'ID of the animal' })
   @ApiResponse({ status: 204, description: 'Animal deleted successfully.' })
+  @ApiBadRequestResponse({ status: 404, description: 'Animal not found' })
+  @ApiOperation({ summary: 'Delete animal by ID' })
   async deleteAnimal(@Param('id') id: number): Promise<void> {
     this.service.delete(id);
   }
 
   @Put(':id/sleep')
   @ApiParam({ name: 'id', description: 'ID of the animal' })
   @ApiResponse({ status: 200, description: 'Animal slept successfully.', type: AnimalDto })
+  @ApiProduces(APPLICATION_JSON_MEDIA_TYPE)
+  @ApiBadRequestResponse({ status: 404, description: 'Animal not found' })
+  @ApiOperation({ summary: 'Put animal to sleep' })
   async sleep(@Param('id') id: number, @Body() data: IncreaseAgeDto): Promise<AnimalDto> {
     return await this.service.sleep(id, data.age).then((value) => fromAnimalModelToDto(value));
   }
 
   @Put(':id/eat')
   @ApiParam({ name: 'id', description: 'ID of the animal' })
-  @ApiResponse({ description: 'Animal ate successfully.', type: AnimalDto })
+  @ApiResponse({ status: 200, description: 'Animal ate successfully.', type: AnimalDto })
+  @ApiProduces(APPLICATION_JSON_MEDIA_TYPE)
+  @ApiBadRequestResponse({ status: 404, description: 'Animal not found' })
+  @ApiOperation({ summary: 'Put animal to eat' })
   async eat(@Param('id') id: number, @Body() data: IncreaseWeightDto): Promise<AnimalDto> {
     return await this.service.eat(id, data.weight).then((value) => fromAnimalModelToDto(value));
   }
 
   @Get(':id/speak')
   @ApiParam({ name: 'id', description: 'ID of the animal' })
-  @ApiResponse({ description: 'Animal spoke successfully.', type: String })
+  @ApiResponse({ status: 200, description: 'Animal spoke successfully.', type: String })
+  @ApiBadRequestResponse({ status: 404, description: 'Animal not found' })
+  @ApiOperation({ summary: 'Make animal speak' })
   async speak(@Param('id') id: number): Promise<String> {
```

Comment:
You don't need to create a new promise. You can return it directly from the service.